src/components/promptBuilder/promptBuilder.tsx

```diff
@@ -0,0 +1,311 @@
+      }
+      props.ws.send(message)
+    }
+  }
+
+  function handleNextQuestion() {
+    handleSendInput()
+    setLastInputMessage(null)
+    setIsLoading(true)
+  }
+
+  function handleQuit() {
+    setIsAttemptingToQuit(false)
+    props.onCancel()
+  }
+
+  function handleGeneratePrompts() {
+    handleSendInput()
+    setIsGeneratingPrompts(true)
+    setIsLoading(true)
+    props.onSubmit()
+  }
+
+  function renderMessages() {
+    if (!isGeneratingPrompts) {
+      return (
+        messages.length > 0 &&
+        messages.map((message) => {
+          if (
+            message.sender.id !== props.sender.id &&
+            message.format !== 'promptBuilder'
+          ) {
+            return (
+              <ElementRenderer
+                key={message.id}
+                ws={inputCapturer}
+                sender={props.sender}
+                message={message}
+                supportedElements={props.supportedElements}
+              />
+            )
+          }
+        })
+      )
+    }
+  }
+
+  function renderNextQuestionButton() {
+    return (
+      <Button
+        variant="outlined"
+        endIcon={<Icon name="chevron_right" />}
+        onClick={handleNextQuestion}
+        disabled={!lastInputMessage}
+        data-cy="next-question-button"
+      >
+        Next question
+      </Button>
+    )
+  }
+
+  function renderGenerateButton() {
+    return (
+      <Button
+        variant="contained"
+        color="secondary"
+        onClick={handleGeneratePrompts}
+        disabled={isLoading}
+        data-cy="generate-button"
+      >
+        Generate
+      </Button>
+    )
+  }
+
+  function renderQuitButton() {
+    return (
+      <Button
+        variant="outlined"
+        startIcon={<Icon name="close" />}
+        onClick={() => setIsAttemptingToQuit(true)}
+        data-cy="quit-button"
+      >
+        Quit
+      </Button>
+    )
+  }
+
+  function renderMobileButtons() {
+    return (
+      <>
+        {!isGeneratingPrompts && (
+          <Box className="rustic-prompt-builder-buttons">
+            <Box className="rustic-prompt-builder-buttons-left">
+              {renderQuitButton()}
+              {renderNextQuestionButton()}
+            </Box>
+            {isReadyToGenerate && (
+              <Box className="rustic-prompt-builder-buttons-right">
+                {renderGenerateButton()}
+              </Box>
+            )}
+          </Box>
+        )}
+      </>
+    )
+  }
+
+  function renderDesktopButtons() {
+    return (
+      <>
+        {!isGeneratingPrompts && (
+          <Box className="rustic-prompt-builder-buttons">
+            <Box className="rustic-prompt-builder-buttons-left">
+              {renderQuitButton()}
+            </Box>
+            <Box className="rustic-prompt-builder-buttons-right">
+              {renderNextQuestionButton()}
+              {isReadyToGenerate && renderGenerateButton()}
+            </Box>
+          </Box>
+        )}
+      </>
+    )
+  }
```

Comment:
Is it possible to avoid repetitive code for desktop and mobile buttons? I think you just need to adjust the css styles based on screen sizes.